unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Common/ShowHideAnimator/ShowHideAnimator.cs

```diff
@@ -1,78 +1,80 @@
-using JetBrains.Annotations;
+using DG.Tweening;
 using UnityEngine;
 
-[RequireComponent(typeof(Animator))]
 public class ShowHideAnimator : MonoBehaviour
 {
+    private const float BASE_DURATION = 0.024f;
+
+    public event System.Action<ShowHideAnimator> OnWillFinishHide;
+    public event System.Action<ShowHideAnimator> OnWillFinishStart;
+
     public bool hideOnEnable = true;
     public float animSpeedFactor = 1.0f;
     public bool disableAfterFadeOut;
-    public string visibleParam = "visible";
 
-    private Animator animatorValue;
+    [SerializeField] private CanvasGroup canvasGroup;
```

Comment:
Why we need to serialize the `canvasGroup` field in the inspector if you are already caching it in the `Awake()` method? It could be private directly, no?

Reply:
I dont really know hahaha, probably leftovers from some tests. Gotta revert it.

Reply:
I remembered now, some items were instantiated and worked with without time for the Awake to be called. As a workaround I assigned the reference in the prefab. Im putting a check in the awake to avoid the `GetComponent` if is it assigned.